timesketch/ui/static/components/explore/explore-heatmap-directive.js

```diff
@@ -0,0 +1,152 @@
+/*
+ Copyright 2015 Google Inc. All rights reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+(function() {
+    var module = angular.module('timesketch.explore.heatmap.directive', []);
+
+    module.directive('tsHeatmap', function ($window, timesketchApi) {
+        /**
+         * Heatmap chart for number of events per hour/weekday.
+         * @param sketchId - Sketch ID.
+         * @param filter - Filter object.
+         * @param query - Query string.
+         * @param meta - Events metadata object.
+         */
+        return {
+            restrict: 'E',
+            scope: {
+                sketchId: '=',
+                filter: '=',
+                query: '=',
+                meta: '=',
+                showCharts: '='
+            },
+            require: '^tsSearch',
+            link: function(scope, element, attrs, ctrl) {
+                var AGGREGATION_TYPE = 'heatmap';
```

Comment:
Not sure what is the best style here for this constant. Can it be outside of this function? 
Should it have trailing underscore to indicate private?